network_analysis/configure-subnetwork-trace/src/main/java/com/esri/samples/configure_subnetwork_trace/ConfigureSubnetworkTraceController.java

```diff
@@ -0,0 +1,419 @@
+import javafx.scene.control.TextField;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.CodedValue;
+import com.esri.arcgisruntime.data.CodedValueDomain;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.utilitynetworks.UtilityAssetGroup;
+import com.esri.arcgisruntime.utilitynetworks.UtilityAssetType;
+import com.esri.arcgisruntime.utilitynetworks.UtilityAttributeComparisonOperator;
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategoryComparison;
+import com.esri.arcgisruntime.utilitynetworks.UtilityDomainNetwork;
+import com.esri.arcgisruntime.utilitynetworks.UtilityElement;
+import com.esri.arcgisruntime.utilitynetworks.UtilityElementTraceResult;
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetwork;
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetworkAttribute;
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetworkAttributeComparison;
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetworkSource;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTerminal;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTier;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceAndCondition;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConditionalExpression;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConfiguration;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceOrCondition;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceParameters;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceResult;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceType;
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraversabilityScope;
+
+public class ConfigureSubnetworkTraceController {
+
+  @FXML private CheckBox includeBarriersCheckBox;
+  @FXML private CheckBox includeContainersCheckBox;
+  @FXML private ComboBox<CodedValue> comparisonValuesComboBox;
+  @FXML private ComboBox<UtilityAttributeComparisonOperator> comparisonOperatorsComboBox;
+  @FXML private ComboBox<UtilityNetworkAttribute> comparisonSourcesComboBox;
+  @FXML private TextArea traceConditionsTextArea;
+  @FXML private TextField comparisonValuesTextField;
+
+  private UtilityElement startingLocation;
+  private UtilityNetwork utilityNetwork;
+  private UtilityTerminal startingTerminal;
+  private UtilityTraceConditionalExpression initialExpression;
+  private UtilityTraceConfiguration initialUtilityTraceConfiguration;
+  private UtilityTraceConfiguration utilityTraceConfiguration;
+
+  @FXML
+  public void initialize() {
+
+    try {
+
+      // add a listener to the comparison value text field, so that it only accepts numerical input separated by a decimal
+      comparisonValuesTextField.textProperty().addListener((observable, oldValue, newValue) -> {
```

Comment:
I think this is the particular case is where an int input of either 0 or 1 is used to represent 'false' or 'true', so the values can only be one of those two. We might have to rig the sample so that wen selecting the comparison source `is network controller`, we'd get a drop-down with true/false.

Reply:
Adding a try/catch to deal with the exception is a possible fix. However it still doesn't resolve the fact that the expected input for 'is subnetwork controller' is either 0 or 1